javascript-javascript1-week4/voiceAssistant.js

```diff
@@ -0,0 +1,175 @@
+let nameOfPerson = "";
+let toDoList = [];
+
+const commands = {
+  "Hello my name is": welcomeWithName,
+  "What is my name": getName,
+  Add: addToDo,
+  Remove: removeToDo,
+  "What is on my todo?": getToDo,
+  "What day is it today?": getDate,
+  "What is": getResult,
+  "Set a timer for": setTimer,
+  "Tell me a evening jokes": tellJokes,
+};
+
+function welcomeWithName(command) {
+  const userName = command.replace("Hello my name is", "").trim();
+  if (nameOfPerson) {
+    return `You already mentioned your name is: ${nameOfPerson}`;
+  } else {
+    nameOfPerson = userName;
+    return `Nice to meet you, ${userName}.`;
+  }
+}
+
+function getName() {
+  if (nameOfPerson) {
+    return `Your name is: ${nameOfPerson}`;
+  } else {
+    return `You haven't mentioned your name yet.`;
+  }
+}
+
+function addToDo(command) {
+  const task = command.replace("Add", "").replace("to my todo", "").trim();
+  if (toDoList.includes(task)) {
+    return `The task "${task}" is already added.`;
+  } else {
+    toDoList.push(task);
+    return `${task} added to your todo.`;
+  }
+}
+
+function removeToDo(command) {
+  const task = command.replace("Remove", "").replace("from my todo", "").trim();
+  const taskIndex = toDoList.indexOf(task);
+  if (taskIndex !== -1) {
+    toDoList.splice(taskIndex, 1);
+    return `Removed ${task} from your todo.`;
+  } else {
+    return `${task} is not in your todo list.`;
+  }
+}
+
+function getToDo() {
+  if (toDoList.length === 0) {
+    return "There is nothing in your todo list.";
+  } else if (toDoList.length === 1) {
+    return `There is 1 todo: ${toDoList[0]}.`;
+  } else {
+    const allButLast = toDoList.slice(0, -1).join(", ");
+    const last = toDoList[toDoList.length - 1];
+    return `There are ${toDoList.length} todos: ${allButLast} and ${last}.`;
+  }
+}
+
+function getDate() {
+  const currentDate = new Date();
+  const day = currentDate.getDate();
+  const month = currentDate.toLocaleString("default", { month: "long" });
+  const year = currentDate.getFullYear();
+  return `${day}. of ${month} ${year}`;
+}
+
+function getResult(command) {
+  const expression = command.replace("What is", "").trim();
+
+  let operator;
+  if (expression.includes("+")) {
+    operator = "+";
+  } else if (expression.includes("-")) {
+    operator = "-";
+  } else if (expression.includes("*")) {
+    operator = "*";
+  } else if (expression.includes("/")) {
+    operator = "/";
+  } else {
+    return "Invalid mathematical expression";
+  }
+
+  const [operand1, operand2] = expression.split(operator).map(Number);
+
+  if (isNaN(operand1) || isNaN(operand2)) {
+    return "Invalid mathematical expression";
+  }
+
+  let result;
+  switch (operator) {
```

Comment:
This could have been done in the `if` statement above to make this function smaller and avoid a few conditions. It's a small optimization, but it is something to be mindful of to gain performance in large projects.

Reply:
i used if statement now.

Reply:
My point was not to replace the `switch` with a `if`. I meant to merge both blocks like this:
```js
if (expression.includes("+")) {
    operator = "+";
    const [operand1, operand2] = expression.split(operator).map(Number);
    result = operand1 + operand2;
}
```

Reply:
ok, looking at it.